server/src/main/java/org/elasticsearch/action/admin/cluster/stats/TransportClusterStatsAction.java

```diff
@@ -431,8 +439,8 @@ public Map<String, RemoteClusterStats> getRemoteStats() {
         }
     }
 
-    private static boolean doRemotes(ClusterStatsRequest request) {
-        return request.doRemotes();
+    private boolean doRemotes(ClusterStatsRequest request) {
+        return REMOTE_STATS.get(settings) && request.doRemotes();
```

Comment:
Can we remove this setting and use one the one ccs_telemetry setting? That would make it easier for customers to turn it off. Or is there a scenario where they might want to turn off one part but not the other?

And it might be risky to have two settings since this setting probably needs to stick around forever since users might have to set it explicitly and then forget to unset both and Product would only get partial telemetry even if the customer thought they changed it.

Reply:
It's certainly possible, the reason why I chose to use 2 settings was that these essentially do two different things - the first controls collection (e.g. potentially memory/performance effects) and the other controls the stats output (e.g. potential problems with Kibana or other clients). Generally, one can work without the other, depending on which problem you're trying to deal with (of course if you shut off collection, the stats output will be only partially useful). That said, I do see your point about users forgetting to remove it, and with this setup, if they shut down collection and forget about it, we won't have any real way to know about it.